### PR TITLE
Background class, and 5 examples.

### LINKED ISSUE #1152: Complete and add a minimal BackgroundTexture class
Having a `BackgroundTexture` class solves quite a few use cases that are difficult for most users to implement without having to use shaders. The name can still be debated. It's pretty close to a `RenderTargetTexture`  currently in experimental, so maybe we should look into that relation. The `BackgroundTexture` is just a wrapper over an `arcade.gl.Texture`. 

Some ponts:

* It should not use the texture atlas because using this type is a good way to avoid flooding the atlas with large images
* Default initializer can take a width and height
* `BackgroundTexure.from_file(path)` would load an image from file. Possibly using the texture cache
* Support scrolling texture coordiantes in pixels
* Support rotating texture coordiantes with preserved aspect ratio
* Support enabling and disabling texture repeat
* Support enabling mipmaps (useful if zooming)
* Support setting a `Mat3` directly for finer control over texture coordinate transformations
* It can also work as a render target, so it also needs to be a framebuffer. `with bg.enabled()`
* Should it have the same pixel scale as the window framebuffer?
* Create a default texture shader users can load. It should be compatible with the vertex format in `arcade.gl.geometry`.

More thinking is needed.

### PR DESCRIPTION
**Pull request by DragonMoffon**
The push Implements a background class, and BackgroundTexture, BackgroundGroup, and ParallaxGroup. Plus five examples: static Background, scrolling Background, BackgroundGroup, ParallaxGroup, and Blending between Backgrounds.

These close #1152 and #991

### CODE REVIEW

File: arcade/__init__.py [hunk 1]
```diff
@@ -294,6 +294,11 @@ def configure_logging(level: int = None):
 from .sprite_list import get_sprites_at_exact_point
 from .sprite_list import get_sprites_at_point
 
+from .backgrounds import BackgroundTexture
+from .backgrounds import Background
+from .backgrounds import BackgroundGroup
+from .backgrounds import ParallaxGroup
```
Comment (einarf) [inline arcade/__init__.py]: I think we should stop adding things to the arcade namespace. It's getting way too cluttered. Users having to import from the `arcade.backgrounds` module is fine.

File: arcade/backgrounds.py [hunk 2]
```diff
@@ -0,0 +1,408 @@
+from PIL import Image
+
+from arcade.context import ArcadeContext
+from arcade.window_commands import get_window
+from arcade.resources import resolve_resource_path
+import arcade.gl as gl
+from pyglet.math import Mat3
+
+
+class BackgroundTexture:
```
Comment (einarf) [inline arcade/backgrounds.py]: Possibly break this module into smaller parts so we can build on it easier. `arcade/backgrounsd` directly with an `__init__py` importing the classes located in sub-modules.

File: arcade/backgrounds.py [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+class BackgroundGroup:
+    """
+    If you have many backgrounds which you would like to draw together and move together this can help.
+
+    The pos of the Background Group is independent of each Background pos.
+    The offset of the BackgroundGroup is the same as each background.
+    """
+
+    def __init__(self):
+        self.backgrounds: list[Background] = []
```
Comment (einarf) [inline arcade/backgrounds.py]: It's probably better to keep this as a private member and expose a backgrounds property with proper docstrings.

File: arcade/resources/shaders/background_fs.glsl [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+uniform float blend = 1;
+
+uniform vec2 pos;
+uniform vec2 size;
+uniform vec2 bounds;
+
+out vec4 fragColor;
+
+void main() {
+    vec2 texSize = vec2(textureSize(backgroundTexture, 0));
```
Comment (einarf) [inline arcade/resources/shaders/background_fs.glsl:19]: Implicit conversion from int to float

File: arcade/resources/shaders/background_fs.glsl [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    vec2 adjustedUV = frag_uv * size;
+
+    vec2 adjusted = (pixelTransform * vec3(adjustedUV, 1.0)).xy;
+
+    adjusted = adjusted / texSize;
+    fragColor = texture(backgroundTexture, adjusted, 0);
```
Comment (einarf) [inline arcade/resources/shaders/background_fs.glsl:25]: Bias argument it optional. Just leave it out

**Comment by einarf** [comment]
Some quick intial comments. This actually looks really solid, but I am not 100% sure about the details around what properties should be on the texture and the background itself to give the most flexbility.

Also there are missing some type annotations here, but I think fixing the more obvious things and just merging it is a better approach here. Play around with the feature and tweak things instead.

Adding a RGBA color multiplier can also add a lot of flexiblity here instead of just blend. We can really merge this as it is now to be honest. Knitpicking on smaller things that are subject to change around a little bit won't move us forward.

**Comment by DragonMoffon** [comment]
For the colour should, do you think I should apply via multiply or addition? Currently, have it as multiply, but if someone wanted to fade in the color, it would be easier if it was added. 

Also, I have it supporting both normalised and not, but I decide if the colour is normalised by seeing if the sum of the colour is less than or equal to 3, which may cause some confusing bugs. Is there a better way of deciding?

**Comment by einarf** [comment]
* We do color multiplication everywhere in arcade right now, so that would make the most sense
* If you want to support byte RGBA and normalized RGBA color they can have different properties. `color`, `color_normalized`

**Comment by DragonMoffon** [comment]
Some of the functions are still missing docstrings. Which will need to be fixed before 2.7 is out. But otherwise, it should be mergeable.

**Pull request merged**